moveit_setup_assistant/src/tools/moveit_config_data.cpp

```diff
@@ -554,6 +554,213 @@ bool MoveItConfigData::outputFakeControllersYAML(const std::string& file_path)
+  for (int i = 0; i < srdf_->groups_.size(); ++i)
+  {
+    emitter << YAML::BeginMap;
+    emitter << YAML::Key << "name";
+    emitter << YAML::Value << srdf_->groups_[i].name_ + "_controller";
+    emitter << YAML::Key << "joints";
+    emitter << YAML::Value << YAML::BeginSeq;
+
+    // Iterate through the joints
+    for (int j = 0; j < groups[i].size(); j++)
+    {
+      emitter << groups[i][j];
+    }
+    emitter << YAML::EndSeq;
+    emitter << YAML::EndMap;
+  }
+  emitter << YAML::EndSeq;
+
+  emitter << YAML::Key << "constraints";
+  emitter << YAML::Value << YAML::BeginMap;
+  emitter << YAML::Key << "goal_time";
+  emitter << YAML::Value << "5.0";
+
+  emitter << YAML::Key << "groups";
+  emitter << YAML::Value << YAML::BeginSeq;
+
+  // Loop through groups
+  for (int i = 0; i < srdf_->groups_.size(); ++i)
+  {
+    emitter << YAML::BeginMap;
+    emitter << YAML::Key << "name";
+    emitter << YAML::Value << srdf_->groups_[i].name_ + "_controller";
+    emitter << YAML::Key << "joints";
+    emitter << YAML::Value << YAML::BeginSeq;
+
+    // Iterate through the joints
+    for (int j = 0; j < groups[i].size(); j++)
+    {
+      emitter << YAML::BeginMap;
+      emitter << YAML::Key << groups[i][j] << YAML::Value << YAML::BeginMap;
+      emitter << YAML::Key << "trajectory";
+      emitter << YAML::Value << "0.60";
+      emitter << YAML::Key << "goal";
+      emitter << YAML::Value << "0.15";
+      emitter << YAML::EndMap;
+      emitter << YAML::EndMap;
+    }
+    emitter << YAML::EndSeq;
+    emitter << YAML::EndMap;
+  }
+  emitter << YAML::EndSeq;
+
+  emitter << YAML::EndMap;
+  emitter << YAML::EndMap;
+  emitter << YAML::EndMap;
+
+  emitter << YAML::Newline;
+  emitter << YAML::Comment("Group Position Controllers");
+  emitter << YAML::Comment("Allows to send single ROS msg of Float64MultiArray to all joints");
+
+  emitter << YAML::BeginMap;
+  emitter << YAML::Key << "joint_position_controller";
+  emitter << YAML::Value << YAML::BeginMap;
+  emitter << YAML::Key << "type";
+  emitter << YAML::Value << "position_controllers/JointGroupPositionController";
+
+  emitter << YAML::Key << "groups";
+  emitter << YAML::Value << YAML::BeginSeq;
+
+  // Loop through groups
+  for (int i = 0; i < srdf_->groups_.size(); ++i)
+  {
+    emitter << YAML::BeginMap;
+    emitter << YAML::Key << "name";
+    emitter << YAML::Value << srdf_->groups_[i].name_ + "_controller";
+    emitter << YAML::Key << "joints";
+    emitter << YAML::Value << YAML::BeginSeq;
+
+    // Iterate through the joints
+    for (int j = 0; j < groups[i].size(); j++)
+    {
+      emitter << groups[i][j];
+    }
+    emitter << YAML::EndSeq;
+    emitter << YAML::EndMap;
+  }
+
+  emitter << YAML::EndSeq;
+  emitter << YAML::EndMap;
+  emitter << YAML::EndMap;
+
+  std::ofstream output_stream(file_path.c_str(), std::ios_base::trunc);
+  if (!output_stream.good())
+  {
+    ROS_ERROR_STREAM("Unable to open file for writing " << file_path);
```

Comment:
use ROS_ERROR_STREAM_NAMED

Reply:
Why do you recommend using the Named version here?
All the functions here that cant output a file uses this
`ROS_ERROR_STREAM("Unable to open file for writing " << file_path);`

Reply:
See the style guidelines under "ROS":
http://moveit.ros.org/documentation/contributing/code/

It makes filtering all the command line output much easier and cleaner. All of the MSA should be updated to use NAMED, but that belongs in a separate simple PR that doesn't clutter this one.

Reply:
Okay I understand, Thanks.